sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java

```diff
@@ -1587,21 +1573,23 @@ public VirtualMachineScaleSetImpl withoutUserAssignedManagedServiceIdentity(Stri
     //
     @Override
     protected void beforeCreating() {
-        if (this.extensions.size() > 0
-            && this.innerModel() != null
-            && this.innerModel().virtualMachineProfile() != null) {
-            this
-                .innerModel()
-                .virtualMachineProfile()
-                .withExtensionProfile(new VirtualMachineScaleSetExtensionProfile())
-                .extensionProfile()
-                .withExtensions(innersFromWrappers(this.extensions.values()));
-        }
+        setExtensions();
     }
 
     @Override
     protected Mono<VirtualMachineScaleSetInner> createInner() {
-        if (isInCreateMode()) {
+        // support flexible vmss with no profile
+        if (this.orchestrationMode() == OrchestrationMode.FLEXIBLE && this.profileNotSet) {
+            this.innerModel().withVirtualMachineProfile(null);
+            return Mono.just(this)
+                .flatMap(virtualMachineScaleSet -> this
+                    .manager()
+                    .serviceClient()
+                    .getVirtualMachineScaleSets()
+                    .createOrUpdateAsync(resourceGroupName(), name(), innerModel()));
```

Comment:
This would be just
```
manager()
                    .serviceClient()
                    .getVirtualMachineScaleSets()
                    .createOrUpdateAsync(resourceGroupName(), name(), innerModel()))
```
?

Reply:
Of course!

Reply:
Refactored the code.